src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/DeclareFee.tsx

```diff
@@ -16,34 +15,41 @@ import { useStores } from '~app/hooks/useStores';
 import { SingleOperator } from '~app/model/processes.model';
 import { getIsContractWallet } from '~app/redux/wallet.slice';
 import { getFromLocalStorageByKey, saveInLocalStorage } from '~root/providers/localStorage.provider';
+import { fetchAndSetOperatorFeeInfo, getFeeIncreaseAndPeriods, getOperatorFeeData, getOperatorProcessId } from '~app/redux/operator.slice.ts';
+import { updateOperatorFee } from '~root/services/operatorContract.service.ts';
 
 const DeclareFee = ({ newFee, oldFee, currentCurrency, getCurrentState }: IncreaseFlowProps) => {
   const stores = useStores();
   const classes = useStyles({});
   const [isLoading, setIsLoading] = useState(false);
-  const operatorStore: OperatorStore = stores.Operator;
   const dispatch = useAppDispatch();
   const processStore: ProcessStore = stores.Process;
   const process: SingleOperator = processStore.getProcess;
   const operator = process.item;
   const isContractWallet = useAppSelector(getIsContractWallet);
+  const processOperatorId = useAppSelector(getOperatorProcessId);
+  const feeIncreaseAndPeriods = useAppSelector(getFeeIncreaseAndPeriods);
+  const operatorFeeData = useAppSelector(getOperatorFeeData);
 
   const changeOperatorFee = async () => {
     setIsLoading(true);
-    const response = await operatorStore.updateOperatorFee({ operator, newFee, isContractWallet, dispatch });
-    await operatorStore.syncOperatorFeeInfo(operatorStore.processOperatorId);
+    const response = await updateOperatorFee({ operator, newFee, isContractWallet, dispatch });
+    await dispatch(fetchAndSetOperatorFeeInfo(processOperatorId));
     if (response) {
       // @ts-ignore
       let savedOperator = JSON.parse(getFromLocalStorageByKey('expired_operators'));
-      if (savedOperator && savedOperator?.includes(operatorStore.processOperatorId)) {
-        savedOperator = savedOperator.filter((item: any) => item !== operatorStore.processOperatorId);
+      if (savedOperator && savedOperator?.includes(processOperatorId)) {
+        savedOperator = savedOperator.filter((item: any) => item !== processOperatorId);
         saveInLocalStorage('expired_operators', JSON.stringify(savedOperator));
       }
     }
-    getCurrentState();
     setIsLoading(false);
   };
 
+  useEffect(() => {
+    getCurrentState();
+  }, [JSON.stringify(operatorFeeData)]);
```

Comment:
relying on property order might not be consistent, if possible, try to relay on something else

Reply:
+